### ISSUE TITLE
False positive for CKV_AZURE_43: check storage account name

### ISSUE THREAD

**Issue by geertn** [bridgecrewio/checkov#741]
I'm building my Storage Account names like this
```
name                = "${local.saname_prefix}diagnostics${module.tf-var-project.random_id}
```

With https://github.com/bridgecrewio/checkov/pull/429 merged I now get a Check failure on the SA name:

```
Check: CKV_AZURE_43: "Ensure the Storage Account naming rules"
	FAILED for resource: azurerm_storage_account.diagnostics
	File: /az_diag_sa.tf:8-22

		8  | resource "azurerm_storage_account" "diagnostics" {
		9  |   #checkov:skip=CKV_AZURE_35:Public access is allowed
		10 |   name                = "${local.saname_prefix}diagnostics${module.tf-var-project.random_id}"

````

**Comment by jasoncastonguay** [comment]
I am also now getting what I believe are similar false positives for CKV_AZURE_43.

```
Check: CKV_AZURE_43: "Ensure the Storage Account naming rules"
	FAILED for resource: azurerm_storage_account.state
	File: /main.tf:27-68

		27 | resource "azurerm_storage_account" "state" {
		28 |   name                      = local.storage_acct
```

The local is defined with
```
   clean_app           = "${substr(replace(lower("${var.app}"), "/[^\\w]/", ""), 0, 16)}${random_string.suffix.result}"
   storage_acct        = local.clean_app
```

Another is:
```
Check: CKV_AZURE_43: "Ensure the Storage Account naming rules"
	FAILED for resource: azurerm_storage_account.bootdiag
	File: /bootdiag.tf:10-46

		10 | resource "azurerm_storage_account" "bootdiag" {
		11 |   name                      = "diag${random_id.randomId.hex}"
```

**Comment by JoonatanNguyen** [comment]
I have the same issue that the storage account name is correct with the rule but it still shows the error that it's failed. 
However, when I run the step in CI pipeline, it showed a failed error but the step is still passed like the picture given bellow. Is this a bug or not, can you confirm this? 
![image](https://user-images.githubusercontent.com/36813718/104590270-c322d600-5673-11eb-8824-a3fb14767856.png)

**Comment by metahertz** [comment]
Hi @JoonatanNguyen Are you using the github action for your CI? Can you re-test the "pass vs fail by default" behaviour?

**Comment by JamesWoolfenden** [comment]
i can replicate this one @metahertz ill show u test code tomox

**Comment by JamesWoolfenden** [comment]
@JoonatanNguyen  the rule triggers id your name is short than 3 or longer than 20? is your storage account name too long?

**Comment by JoonatanNguyen** [comment]
> @JoonatanNguyen the rule triggers id your name is short than 3 or longer than 20? is your storage account name too long?

I checked and counted manually quite many times and it in the range of 3 - 20 characters for the storage account name 🤔

**Comment by JamesWoolfenden** [comment]
yeah im getting an issue, but i thought id at least ask the obvious. i've made those mistakes 👀

**Comment by fculpo** [comment]
Hi, 
I'm also having this issue, my storageaccount is between 3 and 24 characters and terraforms creates it just fine.
However, checkov keeps raising CKV_AZURE_43

**Comment by MaxiPalle** [comment]
Problem still exists with version 1.0.862.

**Comment by daviddob** [comment]
Can confirm the same behavior with the false positive. I have the following variable for name which ensures the same regex validation

```
variable "name" {
  description = "Storage account name"
  type = string
  validation {
    condition     = can(regex("^[a-z0-9]{3,24}$", var.name))
    error_message = "Storage account names must be between 3 and 24 characters in length and may contain numbers and lowercase letters only and must be globally unique."
  }
}
```

however when passing it in as follows, CKV_AZURE_43 is raised
```
resource "azurerm_storage_account" "stg" {
  name                      = var.name
...
}
```

though hardcoding it to something that passes the regex satisfies the condition.
```
resource "azurerm_storage_account" "stg" {
  name                      = "foo"
 ...
 }
 ```

It is likely validating the literal `var.name` instead of a potential variable and as such because `var.name` the literal has a `.` it will fail to satisfy the regex and raise the error. Ideally the check would have to evaluate variables (and determine what do if no default exists) and see if they satisfy the constraint, however that may be a bit much to ask.

**Comment by Richard-CDS** [comment]
As with @daviddob, I too am experiencing the same issue having used variables to construct the storage account name (23 chars long).  

The accounts are created fine, but the rule check fails.
